.drone.jsonnet

```diff
@@ -0,0 +1,81 @@
+local step(name, commands, image='golang:1.13.10') = {
+  name: name,
+  commands: commands,
+  image: image,
+};
+
+local pipeline(name, steps=[]) = {
+  kind: 'pipeline',
+  type: 'docker',
+  name: name,
+  steps: [step('runner identification', ['echo $DRONE_RUNNER_NAME'], 'alpine')] + steps,
+  trigger+: {
+    branch+: ['master'],
+    event+: ['push','pull_request'],
+  },
+};
+
+local masterOnly = {
+  when: {branch:['master'], event: ['push']},
+};
+
+local prOnly = {
+  when: {event: ['pull_request']},
+};
+
+local repo = 'grafana/synthetic-monitoring-agent';
+
+local vault_secret(name, vault_path, key) = {
+  kind: 'secret',
+  name: name,
+  get: {
+    path: vault_path,
+    name: key,
+  },
+};
+
+[
+  pipeline('build', [
+    step('lint', ['make lint']),
+    step('test', ['make test']),
+    step('build', [
+      'git fetch origin --tags',
+      './scripts/version',
+      './scripts/version > .tags', // save version in special file for docker plugin
+      'make build',
+    ]),
+    // We can't use 'make docker' without making this repo priveleged in drone
+    // so we will use the native docker plugin instead for security.
+    step('docker build',[],'plugins/docker')+{
+      settings:{
+        repo: repo,
+        dry_run: 'false',
```

Comment:
@mem my intention was that the docker build step could run on all prs and things, and the docker push step would only run on master. By setting dry_run:false here, you will be pushing to docker hub on each pr build and stuff. Is that what you intended?

Reply:
no, I misread the config and thought this section applied to master. Reverted.

But it did fail and I was expecting it not to. Am I missing something else?

Reply:
I found what I was missing, sorry about the noise!